docs/src/pages/docs/crocks/Result.md

```diff
@@ -0,0 +1,1422 @@
+
+const { Err, Ok } = Result
+
+// gte :: Number -> Number -> Boolean
+const gte =
+  y => x => x >= y
+
+// lte :: Number -> Number -> Boolean
+const lte =
+  y => x => x <= y
+
+// between :: (Number, Number) -> Boolean
+const between = (x, y) => and(gte(x), lte(y))
+
+// ensure :: (a -> bool) -> (a -> Result a a)
+const ensure = pred => ifElse(pred, Ok, Err)
+
+// inRange :: Number -> Result
+const inRange = ensure(between(10, 15))
+
+inRange(12)
+//=> Ok 12
+
+inRange(25)
+//=> Err 25
+
+inRange('Test')
+//=> Err "Test"
+
+const prop =
+  curry((name, obj) => obj[name])
+
+const tryGetAge = compose(
+  bimap(constant('Age not found!'), identity),
+  chain(ensure(isDefined)),
+  tryCatch(prop('age'))
+)
+
+tryGetAge({ name: 'Sarah', age: 23 })
+//=> Ok 23
+
+tryGetAge({ name: 'Sarah' })
+//=> Err "Age not found!"
+
+tryGetAge(1)
+//=> Err "Age not found!"
+
+tryGetAge(undefined)
+//=> Err "Age not found!"
+```
+
+<article id="topic-implements">
+
+## Implements
+`Setoid`, `Semigroup`, `Functor`, `Alt`, `Apply`, `Traversable`,
+`Chain`, `Applicative`, `Monad`
+
+</article>
+
+<article id="topic-construction">
+
+## Construction
+
+```haskell
+Result :: a -> Result e a
+```
+
+Most of the time, `Result` is constructed using functions of your own making
+and helper functions like [`tryCatch`](#trycatch) or by employing one of the
+instance constructors, [`Ok`](#ok) or [`Err`](#err). This is due to the nature
+of `Result` and most other Sum Types.
+
+As a matter of consistency and completion, a `Result` instance can also be
+constructed using its TypeRep like any other type. The `Result` constructor is
+a unary function that accepts any type `a` and returns a [`Ok`](#ok) instance,
+wrapping the value passed to its argument.
+
+```javascript
+import Result from 'crocks/Result'
+
+import equals from 'crocks/pointfree/equals'
+
+const { Ok, Err, of } = Result
+
+Result('some string')
+//=> Ok "some string"
+
+Result(null)
+//=> Ok null
+
+Result(undefined)
+//=> Ok undefined
+
+of('some string')
+//=> Ok "some string"
+
+of(null)
+//=> Ok null
+
+of(undefined)
+//=> Ok undefined
+
+Ok('some string')
+//=> Ok "some string"
+
+Ok(null)
+//=> Ok null
+
+Ok(undefined)
+//=> Ok undefined
+//=> Ok undefined
+
+Err('some string')
+//=> Err "some string"
+
+Err(null)
+//=> Err null
+
+Err(undefined)
+//=> Err undefined
+
+equals(
+  Result.Ok([ 1, 2, 3 ]),
+  Result.of([ 1, 2, 3 ])
+)
+//=> true
+
+equals(
+  of({ a: 100 }),
+  Result({ a: 100 })
+)
+//=> true
+```
+
+</article>
+
+<article id="topic-constructor">
+
+## Constructor Methods
+
+#### Err
+
+```haskell
+Result.Err :: e -> Result e a
+```
+
+Used to construct an [`Err`](#err) instance that represents the "false" or
+"Negative" portion of a disjunction. When an instance is an [`Err`](#err), most
+`Result` returning methods will just return a new [`Err`](#err) instance with
+the same containing value.
+
+The power of the [`Err`](#err) as opposed to a [`Nothing`] is that it can hold
```

Comment:
`Result` is better compared to `Either` instead of `Maybe`. May want to reword this to include the bits about accumulation of `Err` of `Semigroup`s for `alt` and `ap`. Basically, if the reader does not need this accumulation, they should stick with `Either`